src/consensus/blockstore/slot_block_data.rs

```diff
@@ -58,6 +32,18 @@ pub struct SlotBlockData {
     pub(super) equivocated: bool,
 }
 
+#[derive(Debug)]
+pub(super) enum AddShredResult {
+    FirstShred(Slot),
+    Block { slot: Slot, block_info: BlockInfo },
+    Equivocation(Slot),
+    Ok,
+    InvalidProof,
+    InvalidSignature,
+    InvalidShred,
+    Duplicate,
```

Comment:
We don't really handle different error scenarios separately.  So I propose that we just return a `Error` here.